h2o-algos/src/main/java/hex/generic/GenericModel.java

```diff
@@ -52,34 +55,79 @@ private static MojoModel reconstructMojo(ByteVec mojoBytes) {
                 return new ModelMetricsOrdinal.MetricBuilderOrdinal(_output.nclasses(), domain);
             case Regression:  return new ModelMetricsRegression.MetricBuilderRegression();
             case Clustering:
-                assert _mojoModel instanceof KMeansMojoModel;
-                KMeansMojoModel kMeansMojoModel = (KMeansMojoModel) _mojoModel;
-                return new ModelMetricsClustering.MetricBuilderClustering(_output.nfeatures(), kMeansMojoModel.getNumClusters());
+                if (_mojoModel instanceof KMeansMojoModel) {
+                    KMeansMojoModel kMeansMojoModel = (KMeansMojoModel) _mojoModel;
+                    return new ModelMetricsClustering.MetricBuilderClustering(_output.nfeatures(), kMeansMojoModel.getNumClusters());
+                } else {
+                    return unsupportedMetricsBuilder();
+                }
             case AutoEncoder:
                 return new ModelMetricsAutoEncoder.MetricBuilderAutoEncoder(_output.nfeatures());
             case DimReduction:
-                throw new UnsupportedOperationException("DimReduction is not supported.");
+                return unsupportedMetricsBuilder();
             case WordEmbedding:
-                throw new UnsupportedOperationException("WordEmbedding is not supported.");
+                return unsupportedMetricsBuilder();
             case CoxPH:
-                throw new UnsupportedOperationException("CoxPH is not supported.");
-            case AnomalyDetection: return new ModelMetricsAnomaly.MetricBuilderAnomaly("");
-
-            default: throw H2O.unimpl();
+                return unsupportedMetricsBuilder();
+            case AnomalyDetection:
+                return new ModelMetricsAnomaly.MetricBuilderAnomaly("");
+            default:
+                throw H2O.unimpl();
+        }
+    }
+    
+    private ModelMetrics.MetricBuilder unsupportedMetricsBuilder() {
+        if (_parms._disable_algo_check) {
+            Log.warn("Using dummny metric builder for unsupported model category: " + _output._modelCategory);
```

Comment:
Model category 'XY' is currently unsupported. Model metrics will not be available.

Reply:
Done, thanks